frontend/src/Editor/Editor.jsx

```diff
@@ -1302,6 +1300,35 @@ class EditorComponent extends React.Component {
     );
   };
 
+  clonePage = (pageId) => {
+    const currentPage = this.state.appDefinition.pages[pageId];
+    const newPageId = uuid();
+    const newPage = {
+      ...currentPage,
```

Comment:
```suggestion
      ..._.cloneDeep(currentPage),
```
Otherwise, a CSA from one page is running the copy of the CSA on the other page, probably due to same reference being passed in the depths of `currentPage`

Reply:
This is just a guess, please test once you make the change.